test/operand_pattern_test.cpp

```diff
@@ -199,20 +199,29 @@ TEST(AlternatePatternFollowingImmediate, MultipleNonResultIds) {
 TEST(AlternatePatternFollowingImmediate, ResultIdFront) {
   EXPECT_THAT(spvAlternatePatternFollowingImmediate(
                   {SPV_OPERAND_TYPE_RESULT_ID, SPV_OPERAND_TYPE_ID}),
-              Eq(spv_operand_pattern_t{SPV_OPERAND_TYPE_RESULT_ID,
+              Eq(spv_operand_pattern_t{SPV_OPERAND_TYPE_OPTIONAL_CIV,
```

Comment:
These tests are patterns in, patterns out. I chose to fix up just the results rather than reverse both input and output patterns. I believe we still have good coverage of all the places a result id may be, across this test and its two siblings, but exactly what behavior is covered by each test moves around a little. Is that OK?